### PR TITLE
Model specs

### PR DESCRIPTION
**Pull request by NalianAchei2020**
# 📝 Accomplishments
- Use validations for models.
- Write unit tests for models.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -4,6 +4,8 @@ class User < ApplicationRecord
   has_many :likes
   has_many :liked_posts, through: :likes, source: :post
 
+  validates :name, presence: true
+
```
Comment (prolajumokeoni) [inline app/models/user.rb:9]: - [ ] Good job so far, Kindly add validation for your Posts Counter in your User model, from the requirement; `PostsCounter must be an integer greater than or equal to zero`,

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  it 'is valid with a name' do
+    user = User.new(name: 'John Doe')
+    expect(user).to be_valid
+  end
+
+  it 'is invalid without a name' do
+    user = User.new(name: '')
+    expect(user).to_not be_valid
+    expect(user.errors[:name]).to include('can not be blank')
+  end
```
Comment (prolajumokeoni) [inline spec/models/user_spec.rb:51]: - [ ]  It would be nice to add tests for the recent_posts method in your user model 👍🏿 

- [ ] Please also add tests for  your Posts Counter; `PostsCounter must be an integer greater than or equal to zero`, Kindly ensure that all unit  tests written are passing for all your models ✅

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @NalianAchei2020 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ added some validations for models
✔️  wrote some unit tests for all the models




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      5.times { |i| @user.posts.build(title: "Post #{i + 1}") }
+      @user.save
+    end
+
+    it 'returns the most recent posts' do
+      recent_posts = @user.recent_posts
+      expected_count = 0
+      expect(recent_posts.count).to eq(expected_count)
+      expect(recent_posts).to eq(@user.posts.last(expected_count).reverse)
+    end
```
Comment (shahadat3669) [inline spec/models/user_spec.rb]: Great effort so far! As the previous reviewer mentioned, you added tests for the `recent_posts` method in your user spec file, which is a positive step. However, I observed that you're attempting to create five posts for a user, yet you're expecting the count to be 0. Upon closer inspection, it appears that no user or post is being created.

I recommend checking your `User` model file to ensure that you are providing the necessary attributes for user creation. For instance, based on validations, you may need to include the `posts_counter` attribute alongside the `name`. Here's an example:

```ruby
@user = User.create(name: 'Tom', posts_counter: 0)
```

Similarly, review your `Post` model to identify any mandatory attributes required for post creation.

Since you intend to create five posts, the method should ideally return at least three posts. However, it seems that neither the user nor the posts are being created at all. Please revisit your code and make the necessary updates to ensure proper user and post creation. If you encounter any challenges, feel free to seek further assistance. Keep up the good work! 🚀👨‍💻

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @NalianAchei2020, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Worked on previous reviewer comments 
✅ Add test cases for `PostsCounter `



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @NalianAchei2020 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All requested changes implemented :heavy_check_mark: 
- All tests passing and well written :heavy_check_mark: 
- Linters well setup and passing :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
